Fix double click on annotation when annotating an unannotated image

SixK opened the pull request:
Fix double click on annotation when annotating an unannotated image.
There is still a problem when annotate an object, annotate another object with the same annotation id 
then undo this latest annotation.

SixK committed:
Fix double click on annotation when annotating an unannotated image. -
Issue #26

ntjess commented:
Thanks for the PR (and for investigating the issue)! I’ll be able to review and test tomorrow.

SixK committed:
Fix DoubleClick not working on object after a undo.
Add onClick and onDoubleClick callback to undo saved object.
compoundPath.value.clone() probably don't copy callbacks

SixK commented:
Fix for undo added to PR

ntjess commented:
Tested with the following scenarios:
- New annotation on image, focus
- Multiple annotations, focus each one
- annotate, save, refresh, focus
- annotate, undo, focus on prior annotation
- Move rect annotation, focus
- Add metadata to an annotation, focus

It works in all of them! Thanks for the PR.

ntjess commented:
@SixK it seems our CI is stuck which is blocking the merge... Can you try [this suggestion](https://stackoverflow.com/a/58190576) to re-trigger CI?

SixK commented:
forced a new empty push...

jdrakes committed:
Merge branch 'main' into fix-doubleclick

jdrakes commented:
> @SixK it seems our CI is stuck which is blocking the merge... Can you try [this suggestion](https://stackoverflow.com/a/58190576) to re-trigger CI?

@ntjess I went ahead and approved the GH actions to run. For first time contributions to a repo GH doesn't automatically run the GH actions. A safety feature to protect you against bad actors from spamming your resources. I think moving forward for @SixK we won't have to do this. If it happens again I can take a look at the repo protections.
